pkg/importer/vddk-datasource.go

```diff
@@ -53,18 +60,65 @@ func getVddkPluginPath() string {
 	return "vddk"
 }
 
+// NbdOperations provides a mockable interface for the things needed from libnbd.
+type NbdOperations interface {
+	GetSize() (uint64, error)
+	Pread(buf []byte, offset uint64, optargs *libnbd.PreadOptargs) error
+	Close() *libnbd.LibnbdError
+}
+
+// VDDKDataSink provides a mockable interface for saving data from the source.
+type VDDKDataSink interface {
+	Write(buf []byte) (int, error)
+	Close()
+}
+
+// VDDKFileSink writes the source disk data to a local file.
+type VDDKFileSink struct {
+	file   *os.File
+	writer *bufio.Writer
+}
+
+func (sink VDDKFileSink) Write(buf []byte) (int, error) {
```

Comment:
I believe convention wants a pointer, so
```go
func (sink *VDDKFileSink) Write(buf []byte) (int, error) {
```
I would have to look up what the actual practical differences are

Reply:
Okay, I changed this.